README.md

```diff
@@ -1,59 +1,72 @@
+[codecov-url]:https://codecov.io/gh/moxystudio/next-scroll-restoration
+[codecov-image]:https://img.shields.io/codecov/c/github/moxystudio/next-scroll-restoration/master.svg
+[david-dm-url]:https://david-dm.org/moxystudio/next-scroll-restoration
+[david-dm-image]:https://img.shields.io/david/moxystudio/next-scroll-restoration.svg
+[david-dm-dev-url]:https://david-dm.org/moxystudio/next-scroll-restoration?type=dev
+[david-dm-dev-image]:https://img.shields.io/david/dev/moxystudio/next-scroll-restoration.svg
 
-{package-description}
+`next-scroll-restoration`provides a built in custom scroll behavior that aims to make it easy to restore the user scroll position based on location history.
 
 ## Installation
 
 ```sh
-$ npm install @moxy/{package-name}
+$ npm install @moxy/next-scroll-restoration
 ```
 
 This library is written in modern JavaScript and is published in both CommonJS and ES module transpiled variants. If you target older browsers please make sure to transpile accordingly.
 
 ## Motivation
 
-{package-motivation}
+There are some case where you need to take control on how your application scroll is handled, namely you want to restore scroll when user is navigating within your application pages but you need to do extra work before or after the page has changed either by using some sort of page transition or any other feature.
 
-## Usage
+`next-scroll-restoration`provides a built in custom scroll behavior that aims to make it easy to restore the user scroll position based on location history.
 
-{package-usage-example}
+This package is built on top of [scroll-behavior](https://www.npmjs.com/package/scroll-behavior) and it's meant to be used in `Nextjs`applications.
 
-## API
 
-{package-api-description}
+## Usage
+
+Firstly ensure that your `history.scrollRestoration` is set as `manual`within your main App file.
 
-#### {package-api-prop-example}
+```js
+if (typeof history !== 'undefined') {
+    history.scrollRestoration = 'manual';
+}
+```
 
-Type: `object`
-Required: `true`
+Then you can simply import the package and update the scroll whenever your need.
 
-The `{package-api-prop-example}` has the following shape:
 ```js
-{package-api-prop-example}: PropTypes.shape({
-    foo: PropTypes.string,
-    bar: PropTypes.arrayOf(PropTypes.object),
-}).isRequired,
+// ...
+
+import { getScrollBehavior } from '@moxy/next-scroll-behavior';
+
+// ...
+
+const ScrollBehavior = getScrollBehavior();
+
+const updateScrollAfterExtraWork = useCallBack(() => {
+    // ...
+    
+    ScrollBehavior.updateScroll();
+    
+    // ...
+},[])
 ```
 
+## How it works
+
+This package works by actively listening to `Nextjs` router events on`routeChangeStart` and unlistening whenever`routeChangeStart`events off, writing the scroll values associated with the current `location` in the`Session Storage` and reading this values whenever `updateScroll` is called.
```

Comment:
```suggestion
This package works by actively listening to `Nextjs` router events on `routeChangeStart` and unlistening whenever `routeChangeStart` events off, writing the scroll values associated with the current `location` in the `Session Storage` and reading these values whenever `updateScroll` is called.
```

Reply:
Additionally, I think you may have omitted some words from this part:

```
and unlistening whenever `routeChangeStart`events off, [...]
```

I don't really understand what you're trying to convey here.

Reply:
Updated ✅ that sentence was not needed at all.